> I don't think `Locale.ROOT` is a wildcard. Its javadoc says its language is set to the empty string, while a wildcard should have its language set to `*`.

Maybe the docs are a bit confusing, `Locale.ROOT` is a constant, as is `Locale.ENGLISH`.
The relevant code is [here](https://github.com/openjdk/jdk/blob/23fa1a33274d279a53fa6dde683900450561957b/src/java.base/share/classes/sun/util/locale/BaseLocale.java#L45) and [here](https://github.com/openjdk/jdk/blob/7f8450cc511e22e3183092edfa3a37b39f133cff/src/java.base/share/classes/java/util/Locale.java#L656). I did not find another wildcard locale other than `Locale.ROOT`.
> Also, could you add a test for this? It will probably highlight this issue.

This is a good idea; I will try to add a test.

> And I suspect the test failure may be related, somehow, it should be investigated.

Some failing tests seem to be I/O related. For example, one error looks like this:
```
Caused by: io.fabric8.maven.docker.access.hc.http.HttpRequestException: {"message":"Head \"https://registry-1.docker.io/v2/elastic/elasticsearch/manifests/8.15.0\": Get \"https://auth.docker.io/token?account=githubactions&scope=repository%3Aelastic%2Felasticsearch%3Apull&service=registry.docker.io\": net/http: request canceled (Client.Timeout exceeded while awaiting headers)"} (Internal Server Error: 500)
```
Maybe these tests can refactored to be more robust, but this should be handled in another pull request.

--- a/independent-projects/resteasy-reactive/common/runtime/src/main/java/org/jboss/resteasy/reactive/common/headers/HeaderUtil.java
+++ b/independent-projects/resteasy-reactive/common/runtime/src/main/java/org/jboss/resteasy/reactive/common/headers/HeaderUtil.java
@@ -32,7 +32,7 @@ import org.jboss.resteasy.reactive.common.util.WeightedLanguage;
  */
 @SuppressWarnings({ "rawtypes", "unchecked" })
 public class HeaderUtil {
-    private static final List<Locale> LANGUAGE_WILDCARD = List.of(Locale.ROOT);
+    private static final List<Locale> LANGUAGE_WILDCARD = List.of(new Locale("*"));
 
     private static final ClassValue<RuntimeDelegate.HeaderDelegate<?>> HEADER_DELEGATE_CACHE = new ClassValue<>() {
         @Override

--- a/independent-projects/resteasy-reactive/common/runtime/src/test/java/org/jboss/resteasy/reactive/common/headers/HeaderUtilTest.java
+++ b/independent-projects/resteasy-reactive/common/runtime/src/test/java/org/jboss/resteasy/reactive/common/headers/HeaderUtilTest.java
@@ -1,0 +1,28 @@
+package org.jboss.resteasy.reactive.common.headers;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import java.util.Locale;
+
+import org.jboss.resteasy.reactive.common.util.MultivaluedTreeMap;
+import org.junit.jupiter.api.Test;
+
+class HeaderUtilTest {
+
+    @Test
+    void getAcceptableLanguages() {
+        MultivaluedTreeMap<String, String> emptyHeaders = new MultivaluedTreeMap<>();
+        Locale[] wildcardLocale = new Locale[] { new Locale("*") };
+        assertArrayEquals(wildcardLocale, HeaderUtil.getAcceptableLanguages(emptyHeaders).toArray());
+
+        MultivaluedTreeMap<String, String> singleLanguage = new MultivaluedTreeMap<>();
+        singleLanguage.add("Accept-Language", "de");
+        Locale[] singleLocale = new Locale[] { Locale.GERMAN };
+        assertArrayEquals(singleLocale, HeaderUtil.getAcceptableLanguages(singleLanguage).toArray());
+
+        MultivaluedTreeMap<String, String> multipleWeightedLanguages = new MultivaluedTreeMap<>();
+        multipleWeightedLanguages.add("Accept-Language", "da, en-gb;q=0.8, en;q=0.7");
+        Locale[] multipleWeightedLocales = new Locale[] { new Locale("da"), Locale.UK, Locale.ENGLISH };
+        assertArrayEquals(multipleWeightedLocales, HeaderUtil.getAcceptableLanguages(multipleWeightedLanguages).toArray());
+    }
+}

